sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -812,7 +813,7 @@ def test_asin():
     assert asin(p).is_positive is None
     assert asin(sin(S(7)/2)) == -S(7)/2 + pi
     assert asin(sin(-S(7)/4)) == S(7)/4 - pi
-    assert asin(cos(x)) == asin(cos(x))
```

Comment:
How do you know whether this is testing the `__eq__` method or whether it is testing that it not simplified to `x`?

Reply:
In this particular case I know it since I added the test in #16990. In general, it is a problem where the plan is to rely on coverage information, so it is unfortunate that it is messed up here. With that said, I also make a qualified guess before changing it, but not always straightforward to tell.

Reply:
The reason to add the test in #16990 was to increase test coverage, but not for `__eq__`, probably the evaluation.